crates/ruff/src/rules/flake8_todos/rules/todos.rs

```diff
@@ -424,62 +357,19 @@ fn static_errors(
     let after_author = &post_tag[usize::from(author_end)..];
     if let Some(after_colon) = after_author.strip_prefix(':') {
         if after_colon.is_empty() {
+            // TD-005
             diagnostics.push(Diagnostic::new(MissingTodoDescription, tag.range));
         } else if !after_colon.starts_with(char::is_whitespace) {
+            // TD-007
             diagnostics.push(Diagnostic::new(MissingSpaceAfterTodoColon, tag.range));
         }
     } else {
+        // TD-004
         diagnostics.push(Diagnostic::new(MissingTodoColon, tag.range));
 
         if after_author.is_empty() {
+            // TD-005
             diagnostics.push(Diagnostic::new(MissingTodoDescription, tag.range));
         }
     }
 }
-
-#[cfg(test)]
-mod tests {
-    use super::*;
-
-    #[test]
-    fn test_detect_tag() {
-        let test_comment = "# TODO: todo tag";
-        let expected = Tag {
-            content: "TODO",
-            range: TextRange::new(TextSize::new(2), TextSize::new(6)),
-        };
-        assert_eq!(Some(expected), detect_tag(test_comment, TextSize::new(0)));
-
-        let test_comment = "#TODO: todo tag";
-        let expected = Tag {
-            content: "TODO",
-            range: TextRange::new(TextSize::new(1), TextSize::new(5)),
-        };
-        assert_eq!(Some(expected), detect_tag(test_comment, TextSize::new(0)));
-
-        let test_comment = "# todo: todo tag";
-        let expected = Tag {
-            content: "todo",
-            range: TextRange::new(TextSize::new(2), TextSize::new(6)),
-        };
-        assert_eq!(Some(expected), detect_tag(test_comment, TextSize::new(0)));
-        let test_comment = "# fixme: fixme tag";
-        let expected = Tag {
-            content: "fixme",
-            range: TextRange::new(TextSize::new(2), TextSize::new(7)),
-        };
-        assert_eq!(Some(expected), detect_tag(test_comment, TextSize::new(0)));
-        let test_comment = "# noqa # TODO: todo";
-        let expected = Tag {
-            content: "TODO",
-            range: TextRange::new(TextSize::new(9), TextSize::new(13)),
-        };
-        assert_eq!(Some(expected), detect_tag(test_comment, TextSize::new(0)));
-        let test_comment = "# noqa # XXX";
-        let expected = Tag {
-            content: "XXX",
-            range: TextRange::new(TextSize::new(9), TextSize::new(12)),
-        };
-        assert_eq!(Some(expected), detect_tag(test_comment, TextSize::new(0)));
-    }
-}
```

Comment:
Why remove the tests?

Reply:
The `detect_tag` method is removed. To implement the tests on `TodoDirective::from_comment()`, larger changes would be needed (at the minimum `#[derive(Debug...)]` etc., though I was thinking about just moving it from an enum into a struct. Open to making that change and adding unit tests back, though I think the detection logic is pretty thoroughly tested between `T00.py` and `TD00<x>.py`.

Reply:
I generally favor adding `#[derive(Debug)]` to all types. I think spending 15min or so trying to convert the tests to `TodoDirective::from_comment` is worth the time. Debugging unit tests is easier than debugging integration tests.